static/app/utils/replays/hooks/useReplayData.tsx

```diff
@@ -213,21 +210,35 @@ function useReplayData({
     fetchAttachments().catch(onError);
   }, [state.fetchError, fetchAttachments, onError]);
 
+  const clearQueryCache = useCallback(() => {
+    () => {
+      queryClient.invalidateQueries({
+        queryKey: [`/organizations/${orgSlug}/replays/${replayId}/`],
+      });
+      queryClient.invalidateQueries({
+        queryKey: [
+          `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/recording-segments/`,
+        ],
+      });
+      // The next one isn't optimized
+      // This statement will invalidate the cache of fetched error events for replayIds
+      queryClient.invalidateQueries({
+        queryKey: [`/organizations/${orgSlug}/replays-events-meta/`],
+      });
+    };
+  }, [orgSlug, replayId, projectSlug, queryClient]);
+
   return {
     attachments,
     errors,
-    fetchError: state.fetchError,
-    fetching: state.fetchingAttachments || state.fetchingErrors || state.fetchingReplay,
-    onRetry: loadData,
```

Comment:
So it looks like now we invalidate the cache on retry... will it also load the data afterwards?

Reply:
yes. When the cache is invalid, but the hook is still mounted, useQuery will re-fetch.

There are a few related options, like `refetchInterval`, `retryOnMount`, `retryDelay`, and `staleTime` that control fetching as well. None are really being leveraged here now though. https://tanstack.com/query/v4/docs/react/reference/useQuery